pkg/clustermanager/cluster_manager.go

```diff
@@ -252,10 +255,33 @@ func (c *ClusterManager) generateWorkloadKubeconfig(ctx context.Context, cluster
 	return writtenFile, nil
 }
 
-func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, clusterToDelete *types.Cluster) error {
+func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, clusterToDelete *types.Cluster, provider providers.Provider, clusterSpec *cluster.Spec) error {
 	return c.Retrier.Retry(
 		func() error {
-			return c.clusterClient.DeleteCluster(ctx, managementCluster, clusterToDelete)
+			if err := c.PauseEKSAControllerReconcile(ctx, clusterToDelete, clusterSpec, provider); err != nil {
+				return err
+			}
+
+			if err := c.clusterClient.DeleteCluster(ctx, managementCluster, clusterToDelete); err != nil {
+				return err
+			}
+
+			if clusterSpec.GitOpsConfig != nil {
+				if err := c.DeleteGitOpsConfig(ctx, managementCluster, clusterSpec.GitOpsConfig.Name, clusterSpec.GitOpsConfig.Namespace); err != nil {
+					return err
+				}
+			}
+			if clusterSpec.OIDCConfig != nil {
```

Comment:
Do we want to add the same support for AWSIamConfig as well? Since it’s also created as a CR on the cluster.

Reply:
added